### PR TITLE
Add validation logic for ClowdJobInvocation

### PR DESCRIPTION
**Pull request by bsquizz**
Depends on https://github.com/RedHatInsights/clowder/pull/430

Adds validation logic for checking whether a Clowder ClowdJobInvocation has completed, and if it is 'completed', checks to see if any of its invoked child Jobs have failed.

Also follows logs of all jobs invoked by a ClowdJobInvocation

### CODE REVIEW

**Comment by bsquizz** [comment]
Switching back to WIP, need to incorporate job result collection in this area https://github.com/app-sre/qontract-reconcile/blob/a20d254e5e2b3cacfafc26b4aa368b41f7b93962/reconcile/openshift_base.py#L578wt

**Comment by bsquizz** [comment]
Log following added, should be ready for review now

**Comment by Piojo** [comment]
I know they don't exist yet, but can you start a few simple unit tests for `validate_data` while you are at it?

**Comment by maorfr** [comment]
not sure we'd want to block an external contribution for tests we never wrote ourselves. @Piojo are you ok with letting this through without tests?

**Review by maorfr — APPROVED** [review]
LGTM

thanks for your contribution!

**Comment by Piojo** [comment]
Ok, let's move with this PR while we get started with the little refactoring and testing this module needs.

**Review by Piojo — APPROVED** [review]

File: reconcile/openshift_base.py [hunk 1]
```diff
@@ -573,6 +574,27 @@ def validate_data(oc_map, actions):
                         f'({ready_deployments} ready / '
                         f'{managed_deployments} total)')
                     raise ValidationError(name)
+            elif kind == 'ClowdJobInvocation':
```
[the remark this answers is not in the document]
Comment (Piojo) [reply reconcile/openshift_base.py:577]: Let's handle the refactor in a different set of PRs, no point in blocking just this.

**Pull request merged**